mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java

```diff
@@ -114,16 +127,16 @@ private boolean tryMarkAvailable(TaskExecutorID taskExecutorID, TaskExecutorStat
         if (state.isAvailable() && state.getRegistration() != null) {
             TaskExecutorHolder teHolder = TaskExecutorHolder.of(taskExecutorID, state.getRegistration());
             log.debug("Marking executor {} as available for matching.", teHolder);
-            double cpuCores = state.getRegistration().getMachineDefinition().getCpuCores();
-            if (!this.executorByCores.containsKey(cpuCores)) {
-                log.info("[executorByCores] adding {} from TE: {}", cpuCores, teHolder);
-                this.executorByCores.putIfAbsent(
-                    cpuCores,
+            TaskExecutorGroupKey taskExecutorGroupKey = TaskExecutorGroupKey.from(state.getRegistration());
```

Comment:
Shouldn't this be a property of TaskExecutorRegistration if it's purely a function of it?